packages/core/src/modules/ledger/__tests__/IndyPoolService.test.ts

```diff
@@ -0,0 +1,225 @@
+
+      poolService.pools.forEach((pool, index) => {
+        const spy = jest.spyOn(pool, 'submitReadRequest')
+        spy.mockImplementationOnce(responses[index])
+      })
+
+      const { pool } = await poolService.getPoolForDid(did)
+
+      expect(pool.config.id).toBe('indicioMain')
+    })
+
+    it('should return the pool with the self certified did if the did was found on two production ledgers where one did is self certified', async () => {
+      const did = 'VsKV7grR1BUE29mG2Fm2kX'
+      // Found on two production ledgers. Sovrin is self certified
+      const responses = getDidResponsesForDid(did, pools, {
+        sovrinMain: {
+          txnTime: 1632680963,
+          verkey: '~43X4NhAFqREffK7eWdKgFH',
+        },
+        indicioMain: {
+          txnTime: 1632680963,
+          verkey: 'kqa2HyagzfMAq42H5f9u3UMwnSBPQx2QfrSyXbUPxMn',
+        },
+      })
+
+      poolService.pools.forEach((pool, index) => {
+        const spy = jest.spyOn(pool, 'submitReadRequest')
+        spy.mockImplementationOnce(responses[index])
+      })
+
+      const { pool } = await poolService.getPoolForDid(did)
+
+      expect(pool.config.id).toBe('sovrinMain')
+    })
+
+    it('should return the pool with the did with the earlier txnTime if the did was found on two production ledgers where both DIDs are self certified', async () => {
+      const did = 'JHVT5Zv86TrJUJYysET4ij'
+      // Found on two production ledgers. Indicio txnTime is earlier than
+      // Sovrin txnTime
+      const responses = getDidResponsesForDid(did, pools, {
+        sovrinMain: {
+          txnTime: 1632680963,
+          verkey: '~QTQYRnDeYdbo8NDEkWC2Bt',
+        },
+        indicioMain: {
+          txnTime: 1632680000,
+          verkey: '~QTQYRnDeYdbo8NDEkWC2Bt',
+        },
+      })
+
+      poolService.pools.forEach((pool, index) => {
+        const spy = jest.spyOn(pool, 'submitReadRequest')
+        spy.mockImplementationOnce(responses[index])
+      })
+
+      const { pool } = await poolService.getPoolForDid(did)
+
+      expect(pool.config.id).toBe('indicioMain')
+    })
+
+    it('should return the pool with the self certified did if the did was found on two non production ledgers where one did is self certified', async () => {
+      const did = 'HEi9QViXNThGQaDsQ3ptcw'
+      // Found on two non production ledgers. Sovrin is self certified
+      const responses = getDidResponsesForDid(did, pools, {
+        sovrinBuilder: {
+          txnTime: 1632680963,
+          verkey: '~M9kv2Ez61cur7X39DXWh8W',
+        },
+        bcovrinTest: {
+          txnTime: 1632680963,
+          verkey: '3SeuRm3uYuQDYmHeuMLu1xNHozNTtzS3kbZRFMMCWrX4',
+        },
+      })
+
+      poolService.pools.forEach((pool, index) => {
+        const spy = jest.spyOn(pool, 'submitReadRequest')
+        spy.mockImplementationOnce(responses[index])
+      })
+
+      const { pool } = await poolService.getPoolForDid(did)
+
+      expect(pool.config.id).toBe('sovrinBuilder')
+    })
```

Comment:
See these tests for expected outcomes based on which dids are registered on certain ledgers